src/web_interface/components/user_management_routes.py

```diff
@@ -88,45 +87,43 @@ def _change_user_password(self, user_id):
                 flash('password change successful', 'success')
 
     @roles_accepted(*PRIVILEGES['manage_users'])
-    @AppRoute('/admin/edit_user', POST)
-    def ajax_edit_user(self):
-        element_name = request.values['name']
-        if element_name == 'roles':
-            return self._edit_roles()
-        return 'Not found', 400
-
-    def _edit_roles(self):
-        user_name = request.form['pk']
-        selected_role_indexes = sorted(request.form.getlist('value[]'))
-
+    @AppRoute('/admin/edit_user/<user_id>', POST)
+    def ajax_edit_user(self, user_id):
+        if 'input_roles' not in request.form:
+            return 'Bad Request', 400
+        self._edit_roles(user_id)
+        return redirect(url_for('edit_user', user_id=user_id))
+
+    def _edit_roles(self, user_id):
         try:
-            user = self._user_db_interface.find_user(email=user_name)
+            user = self._user_db_interface.find_user(id=user_id)
         except SQLAlchemyError:
-            return 'Not found', 400
+            flash('Error when trying to change user roles')
+            return
+        if user is None:
+            flash(f'Error: user with id {user_id} not found')
+            return
```

Comment:
I'm quite unhappy about the loss of application protocol information here :-/. Would it still be possible to respond with a 500 (`SQLAlchemyError`) and 404 (`user is None`) with the changes in this PR?

Reply:
The endpoint changed from a REST(-like?) endpoint to a "regular" one. So I should probably also remove the "400"

Reply:
merged it with the "change password" functionality into a common POST endpoint